x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/tree/Source.java

```diff
@@ -60,10 +58,8 @@ public static Source readEmpty(StreamInput in) throws IOException {
         if (in.readBoolean() == false) {
             return EMPTY;
         }
```

Comment:
super nit: this is also duplicated between `Source`'s ser/de code and `SourcePosition`'s.

Reply:
I don't particularly want to make an "EMPTY" flavor of `SourcePositions` so I'm going to keep the duplicated boolean read. Seems like the lessor of two evils here.